File: pageobjects/src/test/java/autoTestJavaFullObjects/BaseTests.java
package autoTestJavaFullObjects;

import org.junit.After;
import org.junit.Before;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.UUID;
import java.util.concurrent.TimeUnit;

/**
 * Created by Rafal on 2017-05-31.
 */
public abstract class BaseTests {

    private final String GECKO_DRIVER_NAME = "webdriver.gecko.driver";
    private final String GECKO_DRIVER_PATH = "C:\\bin\\geckodriver.exe";
    protected WebDriver driver;

    @Before
    public void TestSetup() {
        System.setProperty(GECKO_DRIVER_NAME, GECKO_DRIVER_PATH);
        driver = new FirefoxDriver();
        driver.manage().deleteAllCookies();
        driver.manage().timeouts().implicitlyWait(3000, TimeUnit.MILLISECONDS);
        driver.manage().timeouts().pageLoadTimeout(10, TimeUnit.SECONDS);
        driver.manage().window().maximize();
    }

    @After
    public void TestTeardown() {
        if (driver != null) {
            driver.quit();
        }
    }

    protected String generateRandomText() {
        String text = "";
        for (int i=0; i<50; i++) {
            text = UUID.randomUUID().toString() + " " + text;
        }
        return text.trim();
    }

}

File: VisualStudioPageObjects/VisualStudioPageObjects/autoTestJavaFullObjects/BaseTests.cs
using System;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;

namespace autoTestJavaFullObjects
{
    abstract public class BaseTests : IDisposable
    {
        protected IWebDriver driver;

        public BaseTests()
        {
            driver = new ChromeDriver();
            driver.Manage().Cookies.DeleteAllCookies();
            driver.Manage().Timeouts().ImplicitWait = TimeSpan.FromMilliseconds(3000);
            driver.Manage().Timeouts().PageLoad = TimeSpan.FromSeconds(10);
            driver.Manage().Window.Maximize();
        }

        protected string GenerateRandomText()
        {
            string text = "";
            for (int i=0; i<5; i++)
            {
                text = Guid.NewGuid().ToString() + " " + text;
            }
            return text.Trim();
        }

        protected string GenerateRandomEmail()
        {
            return Guid.NewGuid().ToString() + "@test.com";
        }

        protected string GenerateRandomName()
        {
            return Guid.NewGuid().ToString();
        }

        public void Dispose()
        {
            if (driver != null)
            {
                driver.Quit();
            }
        }
    }
}
